src/widgets/Step.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React, { useState, useEffect, memo } from 'react';
+import PropTypes from 'prop-types';
+
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+
+import { SUSSOL_ORANGE } from '../globalStyles';
+import { programStrings, modalStrings } from '../localization';
+
+const getLocalisation = ({ stepKey }) => {
+  const errorString = {
+    program: programStrings.no_programs,
+    supplier: programStrings.no_suppliers,
+    orderType: programStrings.no_order_types,
+    period: programStrings.no_periods,
+  };
+  const placeholders = {
+    program: programStrings.select_a_program,
+    supplier: programStrings.select_a_supplier,
+    orderType: programStrings.select_an_order_type,
+    period: programStrings.select_a_period,
+    name: modalStrings.give_your_stocktake_a_name,
+  };
+
+  return { placeholder: placeholders[stepKey], errorString: errorString[stepKey] };
+};
+
+/**
+ * Component rendering a sequential step in a process, according to it's
+ * current status.
+ * COMPLETE = The step is complete, render a tick and is pressable.
+ * CURRENT = The current step to complete, render an arrow and is pressable.
+ * INCOMPLETE = A future step, no icon and no pressable.
+ * ERROR = A step not able to be completed, render an X and is not pressable.
```

Comment:
Could these be defined in a const enum somewhere?

E.g.

```
const stepStatus = Object.freeze({
  'COMPLETE': 1,
  'CURRENT': 2,
  ...
});
```

Then you can do stuff like:

```
if (status === stepStatus.CURRENT) {
  ...
}
```

Probably a better name, can't think of one off top of my head!


```

Reply:
really like it and went to write it, but it's going to take a few other changes a little out the scope of this PR. Will make this change in the last PR for the issue. Probably making it exportable to use in the `ByProgramModal` component

Reply:
Sounds good!